frontend/src/views/ViewSecretPublicPage/ViewSecretPublicPage.tsx

```diff
@@ -52,8 +62,19 @@ export const ViewSecretPublicPage = () => {
             </a>
           </p>
         </div>
-        {secret && key && <SecretContainer secret={secret} secretKey={key} />}
-        {error && <SecretErrorContainer />}
+        {!isLoading && (
+          <>
+            {!error && !secret && (
+              <PasswordContainer
```

Comment:
I think you have made it much more complex than it should be given the use effect and state. 

What you can do to simplify is. Return a flag in response `isPasswordProtected`. If this flag is true encryptedValue will be null. 

Then after validation with password encryptedValue will be provided. And corresponding ui can be simplified much easier with that way. No need for useEffect

Reply:
The state and effect hooks are used to handle the callback situation as we need to pass the secret from the PasswordContainer child component. Adding a flag is still the same as we have to check if value is `falsy` (this was my initial approach). 

i get the gist of avoiding `effect` and `state` but we need them in this state as secret can be set twice 
- (1). on initial load (is passwordless) 
- (2). from child component if password protected

Reply:
The alternative is to avoid the callback situation and create a parent component for secretContainer, PasswordContainer and secretErrorContainer.